app.go

```diff
@@ -335,6 +344,11 @@ func New(opts ...Option) *App {
 		return app
 	}
 
+	if app.visualize {
+		app.provide(app.DotGraph)
+		app.invokes = append(app.invokes, app.DotGraph) // must be invoked last
```

Comment:
I understand where you are going, but I wonder if there is another way in fx to allow for controller over the order of execution of the dependencies injected. It just feels a bit strange there is an if statement at the end -- which in theory could lead to even more if statements.

Reply:
Removed.